src/process_utility.c

```diff
@@ -1114,6 +1114,46 @@ process_truncate(ProcessUtilityArgs *args)
 							 */
 							list_changed = true;
 					}
+					else if ((chunk = ts_chunk_get_by_relid(relid, false)) != NULL)
+					{
+						ht = ts_hypertable_cache_get_entry(hcache,
+														   chunk->hypertable_relid,
+														   CACHE_FLAG_NONE);
+
+						Assert(ht != NULL);
+
+						/* If the hypertable has continuous aggregates, then invalidate
+						 * the truncated region. */
+						if (ts_continuous_agg_hypertable_status(ht->fd.id) == HypertableIsRawTable)
+						{
+							int64 start = ts_chunk_primary_dimension_start(chunk);
+							int64 end = ts_chunk_primary_dimension_end(chunk);
+
+							Assert(hyperspace_get_open_dimension(ht->space, 0)->fd.id ==
+								   chunk->cube->slices[0]->fd.dimension_id);
+							ts_cm_functions->continuous_agg_invalidate_raw_ht(ht, start, end);
+						}
+
+						/* Truncate the compressed chunk too. */
+						if (chunk->fd.compressed_chunk_id != INVALID_CHUNK_ID)
+						{
+							Chunk *compressed_chunk =
+								ts_chunk_get_by_id(chunk->fd.compressed_chunk_id, false);
+							if (compressed_chunk != NULL)
+							{
+								/* Create list item into the same context of the list. */
+								oldctx = MemoryContextSwitchTo(parsetreectx);
+								rv = makeRangeVar(NameStr(compressed_chunk->fd.schema_name),
+												  NameStr(compressed_chunk->fd.table_name),
+												  -1);
+								MemoryContextSwitchTo(oldctx);
+								list_changed = true;
+							}
+						}
+						list_append = true;
+					}
+					else
+						list_append = true;
```

Comment:
This is horribly hard to follow given the use of `list_append`, `list_changed`, and `rv`. Would benefit from a refactor, but I would accept if you do not think this is in scope for this pull request.

Reply:
Oh yes, agreed. I found it very difficult to understand how this code actually worked.